xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -542,6 +835,8 @@ public void run() {
           }
           logger.log(XdsLogLevel.INFO, "Receive LDS resource update: {0}", update);
           httpMaxStreamDurationNano = update.httpMaxStreamDurationNano;
+          applyFaultInjection = enableFaultInjection && update.hasFaultInjection;
```

Comment:
Still, this had been brought up in the previous PR. I am questioning the `hasFaultInjection` flag from HCM. I do not see any mentioning for HCM has the _"master"_ HttpFilters in the latest design. Also, I don't see C-core's implementation is doing such.

Reply:
In HttpFilter design,
```
Note that gRPC will not fail validation if the map key specifies a filter instance name that does not exist in the HttpConnectionManager filter list.
```

The HCM provides all the filter instance names that can be applied. All downstream overrides with a name that does not exist in the HCM filters will be skipped when merging fault config. The current PR only supports one hard-coded fault filter name, so this `hasFaultInjection` is equivalent to that the name exists in HCM filter instances. 

I agree this flag will be gone after rewriting for the HttpFilter support.